### PR TITLE
Cloudformation template for ch01s04 VPN

### PR DESCRIPTION
**Pull request by antmaro**

### CODE REVIEW

**Review by herve4m — COMMENTED** [review]
Run pre-commit: trailing spaces and YAML lint errors

File: aws/cloudformation/customer-vpc-vpn.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          Value: !Join ["", [!Ref pTagprefix, "-ec2-keypair-customer-vpc-vpn"]]
+        - Key: "redhattraining:course-sku"
+          Value: CS220
+
+  # Security Group by default
+  #rEc2InstanceSecurityGroup:
+  #  Type: AWS::EC2::SecurityGroup
+  #  Properties:
+  #    GroupDescription: Enable SSH access via port 22
+  #    VpcId: !Ref rCustomerVpcVpnPrivateSubnet
```
Comment (herve4m) [inline aws/cloudformation/customer-vpc-vpn.yaml:102]: Yamllint: missing space in comment.
```suggestion
  # rEc2InstanceSecurityGroup:
  #   Type: AWS::EC2::SecurityGroup
  #   Properties:
  #     GroupDescription: Enable SSH access via port 22
  #     VpcId: !Ref rCustomerVpcVpnPrivateSubnet
```

File: aws/cloudformation/customer-vpc-vpn.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        Fn::Select:
+          - 0
+          - Fn::GetAZs: ""
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pTagprefix, "-customer-private-subnet-vpn"]]
+        - Key: "redhattraining:course-sku"
+          Value: CS220
+
+  # New nacl for private subnet
```
Comment (herve4m) [inline aws/cloudformation/customer-vpc-vpn.yaml:64]:
```suggestion
  # NACL for the private subnet
```

File: aws/cloudformation/customer-vpc-vpn.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    Type: AWS::EC2::KeyPair
+    Properties:
+      KeyName: ec2KeyPairVpn
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pTagprefix, "-ec2-keypair-customer-vpc-vpn"]]
+        - Key: "redhattraining:course-sku"
+          Value: CS220
+
+  # Security Group by default
```
Comment (herve4m) [inline aws/cloudformation/customer-vpc-vpn.yaml:97]:
```suggestion
  # Default Security Group for the VPC
```

File: aws/cloudformation/customer-vpc-vpn.yaml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  #rEc2InstanceSecurityGroup:
+  #  Type: AWS::EC2::SecurityGroup
+  #  Properties:
+  #    GroupDescription: Enable SSH access via port 22
+  #    VpcId: !Ref rCustomerVpcVpnPrivateSubnet
+
+  # Create EC2 instance
```
Comment (herve4m) [inline aws/cloudformation/customer-vpc-vpn.yaml:104]:
```suggestion
  # Create an EC2 instance
```

**Review by herve4m — APPROVED** [review]

File: aws/cloudformation/customer-vpc-vpn.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    DependsOn:
+      - rCustomerVpcVpnPrivateSubnet
+      - rNetworkAclCustomerVpcVpnPrivateSubnet
+    Properties:
+      SubnetId:
+        Ref: rCustomerVpcVpnPrivateSubnet
+      NetworkAclId:
+        Ref: rNetworkAclCustomerVpcVpnPrivateSubnet
+
+  #Create new keypair that it will store at /ec2/keypair/key-pair-id
```
[the remark this answers is not in the document]
Comment (antmaro) [reply aws/cloudformation/customer-vpc-vpn.yaml]: It seems pre-commit is complaining that the line too long (94 > 80 characters)  (line-length).

Changing to something like:

`# Keypair is stored in AWS Systems Manager at /ec2/keypair/<key-pair-id>`

**Pull request merged**